### PR TITLE
Stop adding new ops to the graph object with every snapshot and lr change.

### PR DESCRIPTION
**Pull request by ICapalija**
When you call "tf.assign(...)" new tensorflow op is added to the graph object and you are calling it with every snapshot and learning rate update. When graph is growing, memory consumption is increasing until there is no more (and that's a big problem with VGG). You need to define your graph before starting the learning process.

### CODE REVIEW

**Comment by ahmedammar** [comment]
I have included this commit in my pull request #28
